src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -450,11 +446,11 @@ private BoundPattern BindDiscardPattern(DiscardPatternSyntax node, TypeSymbol in
         {
             TypeSyntax typeSyntax = node.Type;
             BoundTypeExpression boundDeclType = BindTypeForPattern(typeSyntax, inputType, diagnostics, ref hasErrors);
-            TypeSymbol declType = boundDeclType.Type;
+            TypeSymbol declType = boundDeclType.Type!;
```

Comment:
>boundDeclType.Type! [](start = 34, length = 19)

How do we know?

Reply:
Not needed.

---
In reply to: [380914909](https://github.com/dotnet/roslyn/pull/41503#discussion_r380914909) [](ancestors = 380914909)